tests/models/test_sync_batchnorm.py

```diff
@@ -1,131 +0,0 @@
-# Copyright The PyTorch Lightning team.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-import pytest
-import torch
-import torch.nn as nn
-import torch.nn.functional as F
-
-from pytorch_lightning import LightningModule, seed_everything, Trainer
-from pytorch_lightning.plugins.environments import LightningEnvironment
-from pytorch_lightning.strategies import DDPSpawnStrategy
-from pytorch_lightning.utilities import FLOAT16_EPSILON
-from tests.helpers.datamodules import MNISTDataModule
-from tests.helpers.runif import RunIf
-from tests.helpers.utils import set_random_main_port
-
-
-class SyncBNModule(LightningModule):
-    def __init__(self, gpu_count=1, **kwargs):
-        super().__init__()
-
-        self.gpu_count = gpu_count
-        self.bn_targets = None
-        if "bn_targets" in kwargs:
-            self.bn_targets = kwargs["bn_targets"]
-
-        self.linear = nn.Linear(28 * 28, 10)
-        self.bn_layer = nn.BatchNorm1d(28 * 28)
-
-    def on_train_start(self) -> None:
-        assert isinstance(self.bn_layer, torch.nn.modules.batchnorm.SyncBatchNorm)
-
-    def forward(self, x, batch_idx):
-        with torch.no_grad():
-            out_bn = self.bn_layer(x.view(x.size(0), -1))
-
-            if self.bn_targets:
-                bn_target = self.bn_targets[batch_idx]
-
-                # executes on both GPUs
-                bn_target = bn_target[self.trainer.local_rank :: self.gpu_count]
-                bn_target = bn_target.to(out_bn.device)
-                assert torch.sum(torch.abs(bn_target - out_bn)) < FLOAT16_EPSILON
-
-        out = self.linear(out_bn)
-
-        return out, out_bn
-
-    def training_step(self, batch, batch_idx):
-        x, y = batch
-
-        y_hat, _ = self(x, batch_idx)
-        loss = F.cross_entropy(y_hat, y)
-
-        return loss
-
-    def configure_optimizers(self):
-        return torch.optim.Adam(self.linear.parameters(), lr=0.02)
-
-
-# TODO: Fatal Python error: Bus error
-@pytest.mark.skip(reason="Fatal Python error: Bus error")
```

Comment:
this test was skipped and outdated (failing for other reasons)
I rewrote it and moved it to parity tests